EVA-1491 — Restructure documentation for the OpenTargets pipeline (#69)
EBIvariation/eva-cttv-pipeline: docs/manual_curation.md

# Manual trait curation

The goal is for traits with occurence ≥ 10 to have 100% coverage after the manual curation. For the rest of the
traits, curate as many as possible.

## Extract information about previous mappings
At this step, mappings produced by the pipeline on the previous iteration (including automated and manual) are
downloaded to be used to aid the manual curation process.

```bash
# Download the latest eva_clinvar release from FTP
wget -qO- ftp://ftp.ebi.ac.uk/pub/databases/eva/ClinVar/latest/eva_clinvar.txt \
  | cut -f4-5 | sort -u > previous_mappings.tsv
```

## Create the final table for manual curation
```bash
python3 bin/trait_mapping/create_table_for_manual_curation.py \
  --traits-for-curation traits_requiring_curation.tsv \
  --previous-mappings previous_mappings.tsv \
  --output table_for_manual_curation.tsv
```

## Sort and export to Google Sheets
Note that the number of columns in the output table is limited to 50, because only a few traits have that many
mappings, and in virtually all cases these mappings are not meaningful. However, having a very large table degrades
the performance of Google Sheets substantially.

```bash

> cut -f-50 table_for_manual_curation.tsv | sort -t$'\t' -k2,2rn | xclip -selection clipboard

Reviewer: make clearer that this won't work in the cluster (xclip)

Author: This is a good point! Actually I think usage of `xclip` here introduces more confusion than it actually helps. I rewrote this section to just indicate that you need to paste the contents of the file to Google Sheets table.

> # Manual trait curation

Reviewer 2: See notes about Markdown style in `docs/build.md`.

Author: Fixed as well

wget -qO- ftp://ftp.ebi.ac.uk/pub/databases/eva/ClinVar/latest/eva_clinvar.txt \
  | cut -f4-5 | sort -u > previous_mappings.tsv
```

## Create the final table for manual curation

```bash
python3 bin/trait_mapping/create_table_for_manual_curation.py \
  --traits-for-curation traits_requiring_curation.tsv \
  --previous-mappings previous_mappings.tsv \
  --output table_for_manual_curation.tsv
```

## Sort and export to Google Sheets

Note that the number of columns in the output table is limited to 50, because only a few traits have that many
mappings, and in virtually all cases these mappings are not meaningful. However, having a very large table degrades
the performance of Google Sheets substantially.

```bash
cut -f-50 table_for_manual_curation.tsv | sort -t$'\t' -k2,2rn | xclip -selection clipboard
```

Create a Google Sheets table like this one
[_traits_requiring_curation_](https://docs.google.com/spreadsheets/d/1mb_ZAEwlSTLCQYBWsihxvUGWoy-otaKFq8tIxpJVT0U/)
and paste the result to that table.

## Manual curation criteria
Good mappings must be eyeballed to ensure they are actually good. Alternative mappings for medium or low quality

> mappings can be searched for using OLS. If a mapping cant be found in EFO, look for a mapping to a HP or ORDO trait

Reviewer 2: ```suggestion
mappings can be searched for using OLS. If a mapping can't be found in EFO, look for a mapping to a HP, ORDO or MONDO trait
```

Author: Fixed the typo

> name. Most HP and ORDO terms will also be in EFO but there are some that arent. These can be imported to EFO using

Reviewer 2: ```suggestion
name. Most HP and ORDO terms will also be in EFO but some are not. These can be imported to EFO using
```

Author: Fixed

the Webulous submission service.

### Criteria to manually evaluate mapping quality

* Exact string for string matches are _good_
* Slight modifications are _good_ e.g. IRAK4 DEFICIENCY -> Immunodeficiency due to interleukin-1  receptor-associated
  kinase-4 deficiency
* Subtype to parent are _good_ e.g ACHROMATOPSIA 3 -> Achromatopsia
* Parent to subtype are _bad_ e.g. HEMOCHROMATOSIS -> Hemochromatosis type 3
Familial / congenital represented on only one half are _bad_ e.g. Familial renal glycosuria -> Renal glycosuria
* Susceptibility on only one half is _bad_ e.g Alcohol dependence, susceptibility to -> alcohol dependence
* Early / late onset on only one half is _bad_ e.g. Alzheimer disease, early-onset -> Alzheimer's disease

> ### Unmapped trait names

Reviewer 2: Same suggestion as in the section above.

Author: Fixed (I presume you meant cant → can't typo fix)

Reviewer 2: That and adding MONDO to the list of ontologies we can get mapping from.

Author: Oh that's right! Fixed in https://github.com/EBIvariation/eva-cttv-pipeline/pull/69/commits/f83186727e96e918c5319312f81b770e1a842b3b

  * 2.1. **The exact mapping in the EFO.** Mark as finished immediately.
  * 2.2. **The exact mapping IS NOT in the EFO.** Similarly to 1.2.1, attempt to find an acceptable EFO mapping; if
    not found, use any acceptable mapping (with subsequent import to EFO).
* 3\. **There are no previously used or exact mappings for the trait.** Curate manually as usual.

### Time-saving options
The new manual workflow can be shortened if necessary, while the quality of the results will be _at least as good as
for the old workflow_ (because we're reusing the results of previous curations):
* All subsections 1.\* involve review of mappings previously selected by ourselves. Because we trust them (to an
  extent), this review can be applied not to all mappings, but only to some (selected on a basis of frequency, or
  just randomly sampled/eyeballed).
* If necessary, section 1 can be skipped completely, i. e. copy-paste previous mappings into “Mapping to use” column,
  but skip the review.
* Sections 2.2 and 3 can only be applied to some variants (e. g. based on frequency), depending on the time
  available.

## Exporting curation results

> Traits with occurence ≥ 10 must have 100% coverage after the manual curation. For the rest of the traits, curate as

Reviewer 2: This first sentence should be at the beginning of the document, so it is always clear how many terms need to be curated.

Author: Moved to the beginning of the document
